src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1335,4 +1360,37 @@ public Object pivot (DataInstance model, EvaluationContext evalContext, List<Obj
         return eval(model, evalContext);
 
     }
+
+    enum HashEncodingMethod {
+        HEX("hex") {
+            @Override
+            String encode(byte[] bytes) {
+                StringBuilder stringBuffer = new StringBuilder();
+                for (byte aResult : bytes)
+                    stringBuffer.append(Integer.toString((aResult & 0xff) + 0x100, 16).substring(1));
+                return stringBuffer.toString();
```

Comment:
Did you consider using the DatatypeConverter for hex as well as base64? The tests pass with this:
`return DatatypeConverter.printHexBinary(bytes).toLowerCase();`

Reply:
Using a javax api felt a little hacky, but after your comment, it's probably ok :)